src/DatePicker/Settings.elm

```diff
@@ -0,0 +1,350 @@
+
+Because it could be the case that a picker is being used in a different
+timezone than the home timezone of the implementor, the `allowedTimesofDay`
+function ingests a `Zone` in addition to a `Posix`. The
+`Zone` represents the time zone in which the picker is being used. An
+implementor can leverage this to compare against a base time zone when
+enforcing allowable times of day, etc. You SHOULD assume that the `Posix`
+passed into these functions is floored to the start of its respective `Day`.
+
+More information can be found in the [examples](https://github.com/mercurymedia/elm-datetime-picker/tree/master/examples).
+
+-}
+type alias TimePickerSettings =
+    { timeStringFn : Zone -> Posix -> String
+    , allowedTimesOfDay :
+        Zone
+        -> Posix
+        ->
+            { startHour : Int
+            , startMinute : Int
+            , endHour : Int
+            , endMinute : Int
+            }
+    }
+
+
+{-| A shared type facilitating the preset variants for both pickers
+-}
+type Preset
+    = PresetDate PresetDateConfig
+    | PresetRange PresetRangeConfig
+
+
+{-| Set type facilitating the preset dates
+`title` - the displayed name of the preset
+`date` - the `Posix` the preset will select in the date picker.
+-}
+type alias PresetDateConfig =
+    { title : String
+    , date : Posix
+    }
+
+
+{-| Set type facilitating the preset date ranges
+`title` - the displayed name of the preset
+`range` - the time range the date range picker will select, consisting of
+a start-`Posix` and an end-`Posix`.
+-}
+type alias PresetRangeConfig =
+    { title : String
+    , range : { start : Posix, end : Posix }
+    }
+
+
+{-| The type facilitating the Theme with the most important design tokens
+-}
+type alias Theme =
+    { fontSize :
+        { base : Css.Px
+        , sm : Css.Px
+        , xs : Css.Px
+        , xxs : Css.Px
+        }
+    , color :
+        { text :
+            { primary : Css.Color
+            , secondary : Css.Color
+            , disabled : Css.Color
+            }
+        , primary :
+            { main : Css.Color
+            , contrastText : Css.Color
+            , light : Css.Color
+            }
+        , background :
+            { container : Css.Color
+            , footer : Css.Color
+            , presets : Css.Color
+            }
+        , action : { hover : Css.Color }
+        , border : Css.Color
+        }
+    , size :
+        { presetsContainer : Css.Px
+        , day : Css.Px
+        , iconButton : Css.Px
+        }
+    , borderWidth : Css.Px
+    , borderRadius :
+        { base : Css.Px
+        , lg : Css.Px
+        }
+    , boxShadow :
+        { offsetX : Css.Px
+        , offsetY : Css.Px
+        , blurRadius : Css.Px
+        , spreadRadius : Css.Px
+        , color : Css.Color
+        }
+    , zIndex : Int
+    , transition : { duration : Float }
+    }
+
+
+{-| The default theme that is included in the defaultSettings
+-}
+defaultTheme : Theme
+defaultTheme =
```

Comment:
Will this default theme look exactly like the previous theme we provided as CSS?

Reply:
No, I made a few changes. Also the Theme itself is structured differently than the CSS-one from before. Is that a problem?

Reply:
Okay. We mention it in the Changelog. So it should be fine